lib/httpServer/transports/http/index.js

```diff
@@ -615,6 +637,48 @@ function safeDecodePath(path) {
 	}
 }
 
+/**
+ * Encapsulate the request object
+ *
+ * This function currently only takes care of
+ * checking the HTTP POST size.
+ *
+ * @param {*} req HTTP request
+ */
+function setRequestFilters(req) {
+	var realOn = req.on.bind(req);
+	var byteSize = 0;
+
+	function isRequestTooLarge() {
+		return byteSize > maxRequestSize;
+	}
+
+	req.on('data', function (chunk) {
+		byteSize += chunk.length;
+
+		if (isRequestTooLarge()) {
+			req.emit('error', new RequestError('Entity too large', req));
```

Comment:
Does this also close the connection? Else we would still be receiving data, just not in userland.

Reply:
Have you tried using `req.destroy(new RequestError('Entity too large', req));`? I think it may be made for this.
I wonder if it would avoid the trickery that's happening below.

Reply:
I have. It doesn't emit the error nor stop the 'data' emission.

Reply:
Could that be worth filing an issue over?

The error should definitely be emitted according to the docs:

> Calls destroy() on the socket that received the IncomingMessage. If error is provided, an 'error' event is emitted and error is passed as an argument to any listeners on the event.

Reply:
Ah, apologies, I believe it does throw the error. The issue was that it also abruptly close the connection, not leaving the chance to external APIs to close/return what they wish. 

At any rate, either way does not stop the 'data' or 'end' data emission, however.

Reply:
I'm missing why abruptly closing the connection is a problem. Isn't that exactly what we would want?

That is why I asked in my initial comment:

> Does this also close the connection?

Reply:
On error, upper levels can decide how to respond on their own terms.

Calling destroy returns *no response*, it just drops the connection.

Reply:
Alright, I see what you're doing. I thought the premise from the beginning *was* to clear out the connection completely, with the assumption that data uploads that are wildly beyond an application's limits would be regarded abuse of some sort.

Reply:
Since we are doing the filtering for *all* input on HTTP POST data, the idea is to let higher levels decide what error they wish to return. The idea behind this is that the communication protocol remains unknown to the HTTP server itself at this point (do we speak JSON? how should an error be returned?).

Reply:
My idea of the right way to avert an attack or abuse of some sort would be to cut the connection and not care about communicating anything back to the client. We do not have a responsibility to clients with ill intentions. We can still log it server-side.

But if there are use-cases where you really want to be friendly to the client, then yeah, we would have to be a bit more friendly. In that case, HTTP status code 413 was made for this and would make for a fine default. I would let the HTTP status code take care of the communication, not the payload you send back (just don't send any).

Reply:
Once the server sends a response, I understand that the upload will be interrupted anyway. Sending an error response that will be supported by the underlying communication protocol off the bat only adds minimal overhead, and makes it easier for protocol implementer to handle such cases without having to worry on top of that about HTTP headers.

Reply:
So then maybe just send back 413 and leave it there? Rather than giving a whole lot of freedom (and responsibility) to user land?

Reply:
I more leaning towards providing a mean for higher levels to respond with an error within the protocol. This should make error management on the client end easier to do.

Reply:
That adds more responsibility to server code though. In this PR, the command center will end up sending a `400 Bad Request` status back. That's less helpful than a `413` imho. So, I understand what you want to see, but then it would be nice if this PR demonstrated its proper use.

Reply:
That structurally relies on requests sent over HTTP and demands that client implementations care about HTTP codes to do a good job. In the current case, I believe clients (and developers using them) won't have to change a single line of code and still receive a meaningful error that does not require them to understand HTTP. This means that they can code against it, which I don't believe they would be able to do should it be HTTP error code based.

Right now, we are sending "server" error though, so that would need to be changed.

Reply:
I traced the logic back to it sending back a `400`? Did I miss something?

Reply:
No, you are right indeed. So we'd need to fix that as well.